objects/src/accounts/storage.rs

```diff
@@ -0,0 +1,50 @@
+use super::{Digest, Hasher, Word, ZERO};
+
+// TYPE ALIASES
+// ================================================================================================
+
+pub type StorageItem = (Word, Word);
+
+// ACCOUNT STORAGE
+// ================================================================================================
+
+/// Account storage is a key-value map where both keys and values are words (4 elements).
+///
+/// Internally, account storage uses a compact Sparse Merkle tree to store the data. To ensure that
+/// the data is randomly distributed across the tree, the index of item in the tree is derived by
+/// hashing the key.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct AccountStorage {
+    // TODO: add compact SMT as backing storage
+}
+
+impl AccountStorage {
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+    /// Returns a new instance of account storage initialized with the provided items.
+    pub fn new(_items: &[StorageItem]) -> Self {
+        todo!()
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns a commitment to this storage.
+    pub fn root(&self) -> Digest {
+        todo!()
+    }
+
+    /// Returns an item from the storage at the specified key.
+    ///
+    /// If the item is not present in the storage, [ZERO, ZERO, ZERO, ZERO] is returned.
```

Comment:
it will probably be beneficial if we specify that the return is a `[Word]`, that is composed of 4 field elements

Reply:
Done!